BPTree.java

```diff
@@ -454,6 +480,37 @@ void rebalance(InternalNode prev) {
         	prev.promote(sibling,this,this.keys.remove(0));
         }
         
+        /**
+         * Creates a new sibling node holding the smaller half of the keys
+         * from the current node
+         * The current node retains all keys from the middle onward
+         * @return new sibling node
+         */
+        Node split() {
+        	
+        	InternalNode sibling = new InternalNode();
+        	int halfPointKey = (keys.size())/2;
+        	int halfPointChild = (children.size())/2;
+        	
+        	//give everything to left of middle key to sibling
+        	for (int i = 0; i < halfPointKey; i++) {
+        		sibling.keys.add(keys.remove(0));

```

Comment:
are you only removing the first key or is this supposed to be .remove(i)   ?

Reply:
This is intentional, I'm removing the first key